src/AddressBook/ABGroup.cs

```diff
@@ -110,14 +112,20 @@ internal ABGroup (NativeHandle handle, ABAddressBook addressbook)
 		}
 
 		public string Name {
-			get {return PropertyToString (ABGroupProperty.Name);}
+			get
+			{
+				var str = PropertyToString (ABGroupProperty.Name);
+				if (str is null)
+					ObjCRuntime.ThrowHelper.ThrowArgumentNullException ("PropertyToString returned null");
+				return str;
+			}
```

Comment:
This as per Microsoft guidelines is a very bad practice: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property?redirectedfrom=MSDN

```
[](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property?redirectedfrom=MSDN#indexed-property-design)❌ AVOID throwing exceptions from property getters.
Property getters should be simple operations and should not have any preconditions. If a getter can throw an exception, it should probably be redesigned to be a method. Notice that this rule does not apply to indexers, where we do expect exceptions as a result of validating the arguments.
```
The correct way probably is to return null and make the string nullable.

Reply:
Ok, considering next line is:

- 	public ABRecord Source {
+ public ABRecord? Source {

We should not throw and return string?

Reply:
Yeah, we have been a little inconsistent since we do have other props that throw but I agree, probably making it nullable would be better in this scenario